### PR TITLE
Reference ECS release candidate images for examples and submodules

### PR DESCRIPTION
**Pull request by Ganeshrockz**
## Changes proposed in this PR:
- Bump ECS image variables to refer to the latest release candidate's name
- Split existing acceptance test matrix into Fargate and EC2 to reduce flakiness

## How I've tested this PR:
CI

## How I expect reviewers to test this PR:
👁️ 

## Checklist:
- [ ] Tests added
- [ ] CHANGELOG entry added 

    [HashiCorp engineers only. Community PRs should not add a changelog entry.]::
    [Changelog entries should use present tense, e.g. "Add support for..."]::

### CODE REVIEW

File: examples/admin-partitions/terraform/variables.tf [hunk 1]
```diff
@@ -20,7 +20,7 @@ variable "tags" {
 variable "consul_ecs_image" {
   description = "Consul ECS image to use."
   type        = string
-  default     = "hashicorpdev/consul-ecs:latest"
+  default     = "hashicorppreview/consul-ecs:0.7.0-rc1"
```
Comment (absolutelightning) [inline examples/admin-partitions/terraform/variables.tf:23]: can we define a variable 
```
variable "default_consul_ecs_image" {
  default = "hashicorppreview/consul-ecs:0.7.0-rc1"
}
```
and use it here like this - 
```
variable "consul_ecs_image" {
  description = "Consul ECS image to use."
  type        = string
  default     = "hashicorpdev/consul-ecs:latest"
  default     = var.default_consul_ecs_image
}
```
or this is not suitable here? I see this repeated lot of times so I am suggesting.

Comment (Ganeshrockz) [reply examples/admin-partitions/terraform/variables.tf:23]: The method you proposed won't work across submodules. In the current state, we define one variable per submodule/terraform example.

Comment (absolutelightning) [reply examples/admin-partitions/terraform/variables.tf:23]: ok got it.

**Review by absolutelightning — APPROVED** [review]
LGTM!

**Pull request merged**